test/pkg.jl

```diff
@@ -139,6 +139,11 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+    Pkg.checkout("Example", "test-branch") #17364
```

Comment:
repo should have remote branch in order to test this part of `checkout` functionality

if sha is provided it's no-brainer, set head to sha